Sources/ActorSingletonPlugin/ActorSingletonPlugin.swift

```diff
@@ -73,16 +96,16 @@ extension ActorSingletonPlugin {
 extension ActorSingletonPlugin: _Plugin {
     static let pluginKey = _PluginKey<ActorSingletonPlugin>(plugin: "$actorSingleton")
 
-    public var key: Key {
+    public nonisolated var key: Key {
         Self.pluginKey
     }
 
     public func start(_ system: ClusterSystem) async throws {}
 
-    public func stop(_ system: ClusterSystem) {
-        self.singletonsLock.withLock {
-            for (_, singleton) in self.singletons {
-                singleton.stop(system)
+    public nonisolated func stop(_ system: ClusterSystem) {
```

Comment:
Arguably should probably become async instead?

Reply:
Yes, though TBH not sure how to handle async in `ClusterSystem.shutdown`.